railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -69,6 +69,9 @@ def gitignore
     def version_control
       if !options[:skip_git] && !options[:pretend]
         run "git init", capture: options[:quiet], abort_on_failure: false
+        if user_default_branch.strip.empty?
+          `git symbolic-ref HEAD refs/heads/main`
```

Comment:
Instead of doing this. I think we should set the -b argument to `git init`.

Reply:
Did you mean switch to `main` using `git checkout -b main`?
To my knowledge, there's no `-b` argument for `git init`.

Reply:
No. I mean switching `git init` to be `git init -b main`. There is since git 2.28.  We could check the version of git and only use that option on git 2.28.

https://git-scm.com/docs/git-init

```
-b <branch-name>
--initial-branch=<branch-name>
Use the specified name for the initial branch in the newly created repository. If not specified, fall back to the default name (currently master, but this is subject to change in the future; the name can be customized via the init.defaultBranch configuration variable).
```